common/oatbox/log/LoggerService.php

```diff
@@ -29,98 +30,104 @@ class LoggerService extends ConfigurableService implements LoggerInterface
     }
 
-    /**
-     * Load the logger from configuration
-     *
-     * If options does not contain any Psr3 Logger, NullLogger is set by default
-     *
-     * @return LoggerInterface
-     */
-    protected function loadLogger()
+    private function loadLoggers(): void
     {
-        $logger = null;
-        if ($this->hasOption(self::LOGGER_OPTION)) {
-            $loggerOptions = $this->getOption(self::LOGGER_OPTION);
-            if (is_object($loggerOptions)) {
-                if (is_a($loggerOptions, LoggerInterface::class)) {
-                    $logger = $loggerOptions;
-                }
-            } elseif (is_array($loggerOptions) && isset($loggerOptions['class'])) {
-                $classname = $loggerOptions['class'];
-                if (is_a($classname, LoggerInterface::class, true)) {
-                    if (isset($loggerOptions['options'])) {
-                        $logger = new $classname($loggerOptions['options']);
-                    } else {
-                        $logger = new $classname();
-                    }
-                }
-            }
+        foreach ($this->getOption(self::OPTION_LOGGERS) ?? [] as $logger) {
+            $this->loadLogger($logger);
+        }
+    }
+
+    private function loadLogger($logger): void
+    {
+        if ($logger instanceof TaoMonolog) {
+            $this->registerLogger($logger, $logger->getName());
+            return;
+        }
+
+        if ($logger instanceof LoggerInterface) {
+            $channel = method_exists($logger, 'getName') ? $logger->getName() : self::DEFAULT_CHANNEL;
+
+            $this->registerLogger($logger, $channel);
+            return;
         }
 
-        if (!is_null($logger)) {
-            $this->logger = $logger;
-        } else {
-            $this->logger = new NullLogger();
+        if (!is_array($logger)) {
+            throw new \LogicException('Logger options must be an array');
         }
 
-        return $this->logger;
+        if (!array_key_exists(self::OPTION_CLASS, $logger)) {
+            throw new \LogicException('No class defined for logger');
+        }
+
+        if (!is_a($logger[self::OPTION_CLASS], LoggerInterface::class, true)) {
+            throw new \LogicException(sprintf('Logger class must implement %s', LoggerInterface::class));
+        }
+
+        $channel = $logger[self::OPTION_OPTIONS][self::OPTION_NAME] ?? self::DEFAULT_CHANNEL;
+
+        $this->registerLogger(new $logger[self::OPTION_CLASS]($logger[self::OPTION_OPTIONS]), $channel);
     }
 
-    /**
-     * Check if the logger is loaded from configuration
-     *
-     * @return bool
-     */
-    protected function isLoggerLoaded()
+    private function registerLogger(LoggerInterface $logger, string $channel): void
     {
-        return $this->logger instanceof LoggerInterface;
+        $this->loggers[$channel] = array_key_exists($channel, $this->loggers)
+            ? new LoggerAggregator([$logger, $this->loggers[$channel]])
```

Comment:
it could cause nested `LoggerAggregator`s while it's possible to read list of loggers from existing `LoggerAggregator`, add new one to it and create new `LoggerAggregator` with flat array of loggers.

Keeping in mind that it's duplicated operation (see my commend above) consider to extract it to the separate method, for example located in `LoggerAggregator` itself:
```
/**
 * Creates new LoggerAggregator with all exising loggers and new $logger
 * @return LoggerAggregator
 */
public function withLogger(LoggerInterface $logger)
```